app/src/main/java/com/sdpteam/connectout/profile/ProfileActivity.java

```diff
@@ -19,47 +22,44 @@
 public class ProfileActivity extends AppCompatActivity {
 
     private final ProfileViewModel pvm = new ProfileViewModel(new ProfileFirebaseDataSource());
+    Authentication auth = new GoogleAuth();
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_profile);
 
-        Profile userProfile;
+        Button editProfile = findViewById(R.id.buttonEditProfile);
+        editProfile.setVisibility(View.INVISIBLE);
 
-        // fetch data
-        AuthenticatedUser au = new GoogleAuth().loggedUser();
-        // id public id
-        String id = getIntent().getStringExtra("id");
-        // user id
-        String uid = (au == null) ? NULL_USER : au.uid;
-
-        if (id != null) {
-            // public user profile
-            userProfile = pvm.getProfile(id).getValue();
-        } else {
-            // current user profile
-            userProfile = pvm.getProfile(uid).getValue();
-            //TODO : what is going on here?
-            // pvm.saveProfile(userProfile);
-
-            // getting the elements references
-            Button editProfile = findViewById(R.id.buttonEditProfile);
-            editProfile.setOnClickListener(v -> goToEditProfile());
+        String userIdToDisplay = getIntent().getStringExtra("uid");
+        if (userIdToDisplay == null) {
+            if (auth.isLoggedIn()) {
+                userIdToDisplay = auth.loggedUser().uid;
+                editProfile.setVisibility(VISIBLE);
+                editProfile.setOnClickListener(v -> goToEditProfile());
+            } else {
+                Log.println(Log.WARN, "ProfileActivity argument exception", "What user do you want to display? Displaying null user for the moment");
+                userIdToDisplay = NULL_USER;
+            }
```

Comment:
No there is no TODO because this it is what we want to do it this case :  display a null user. 
But I wanted show to the programmer a warning that it may not be the profile it wants to display. I will leave this as a Log and yes maybe change the warning message.